### ISSUE TITLE
Clone() vs. copy on a pointer to a tensor attribute. 

### ISSUE THREAD

**Issue by melhindi** [OpenMined/PySyft#4562]
## Description
When I invoke `.clone().get()` on a pointer to a tensor attribute I get an unexpected/wrong value back. Calling `.copy().get()` on the same pointer to a tensor attribute returns the correct data .
Is this behaviour intended? 

## How to Reproduce
Install the required dependencies (syft = {extras = ["udacity"], version = "*"}) and copy&paste the following code into a file.
```
import torch
import sys
from torch import nn
from torch import optim
import syft as sy


def train(datasets, model):
    for iter in range(10):
        for data,target in datasets:
            model.send(data.location)
            opt = optims.get_optim(data.location.id)
            opt.zero_grad()
            pred = model(data)
            loss = ((pred - target)**2).sum()
            loss.backward()
            """
            Begin: Reproducing the issue
            """
            for p in model.parameters():
                print(p.grad.data.copy().get()) # this returns tensor([[0.0000, 0.7492]])
                print(p.grad.data.clone().get()) # this returns tensor([[0.3831, 0.2120]], requires_grad=True)
                # print(p.clone().get().grad.data) # this would return the correct/expected value: tensor([[0.0000, 0.7492]])
                break
            sys.exit() # exit to only show the issue (ie. one iteration is enough)
            """
            END: Reproducing the issue
            """

            opt.step()
            model.get() # Note: this would fail with an ObjectNotFoundError(obj_id, self) although we called clone on the pointer. With copy() it works as expected

            # 6) print our progress
            print(loss.get()) # NEW) slight edit... need to call .get() on loss\
            break
    
# federated averaging

if __name__ == "__main__":

    hook = sy.TorchHook(torch)
    
    bob = sy.VirtualWorker(hook, id="bob")
    alice = sy.VirtualWorker(hook, id="alice")
    
    # A Toy Dataset
    data = torch.tensor([[0,0],[0,1],[1,0],[1,1.]], requires_grad=True)
    target = torch.tensor([[0],[0],[1],[1.]], requires_grad=True)
    
    # get pointers to training data on each worker by
    # sending some training data to bob and alice
    data_bob = data[0:2]
    target_bob = target[0:2]
    
    data_alice = data[2:]
    target_alice = target[2:]
    
    # Iniitalize A Toy Model
    model = nn.Linear(2,1)
    
    data_bob = data_bob.send(bob)
    data_alice = data_alice.send(alice)
    target_bob = target_bob.send(bob)
    target_alice = target_alice.send(alice)
    
    # organize pointers into a list
    datasets = [(data_bob,target_bob),(data_alice,target_alice)]
    from syft.federated.floptimizer import Optims
    workers = ['bob', 'alice']
    optims = Optims(workers, optim=optim.Adam(params=model.parameters(),lr=0.1))
    train(datasets, model)
```
## Expected Behavior
`.clone().get()` and `.copy().get()` should yield the same values, especially if no further state-changing actions are performed (e.g., only one train iteration on a model).

## System Information
 - OS: MacOS Catalina
 - OS Version: 10.15.6
 - Language Version: Python 3.7.5 
 - Package Manager Version: pipenv, version 2020.8.13

## Additional Context
As shown in the the reproducing example I came across this issue, while trying to access the gradients that were computed on the remote side.

**Comment by LaRiffle** [comment]
```python
x = th.tensor([1, 2])
p = x.send(alice)
print(p)
print(p.clone())
print(p.copy())
```
prints
```
(Wrapper)>[PointerTensor | me:34294736491 -> alice:26236999679]
(Wrapper)>[PointerTensor | me:34294736491 -> alice:26236999679]
(Wrapper)>[PointerTensor | me:17174615997 -> alice:9847375338]
``` 
Hi!
`clone`clones the pointer while `copy` copies the remote value, so return a new poitner

**Comment by melhindi** [comment]
Thanks for the clarification @LaRiffle, however, it is not clear to me why the two statements in my example yield different results. Even if I clone the pointer and later call `.get()` it should return the same value, especially if nothing should have happened in the meantime.

**Comment by LaRiffle** [comment]
So yeah I guess it solves your note: `# Note: this would fail with an ObjectNotFoundError(obj_id, self) although we called clone on the pointer. With copy() it works as expected`  because with clone remote copy wasn't duplicated
For the other part:
```
print(p.grad.data.copy().get()) # this returns tensor([[0.0000, 0.7492]])
                print(p.grad.data.clone().get()) # this returns tensor([[0.3831, 0.2120]], requires_grad=True)
                # print(p.clone().get().grad.data) # this would return the correct/expected value: tensor([[0.0000, 0.7492]])
                
```
The .grad.data on pointers might not be 100% stable, if you can copy().get() or clone().get() directly on the pointer it would solve your issue I guess! Alternatively, if you build a minimal example of this error we can try investigating it, although this kind of bugs will disappear in 0.3.
I hope this helps :)

**Comment by cereallarceny** [comment]
Hello! Just letting you know that we are no longer planning on supporting anything on the 0.2.x product line and that all work should be ported over to 0.3.x, which is considered a complete rebuild of PySyft. Because of that, I'll be closing this issue. If you feel this is a mistake, or if the issue actually applies to 0.3.x as well, please feel free to ping me on Slack and I'll reopen the issue.
